source/features/pull-request-timeline-filter.tsx

```diff
@@ -0,0 +1,260 @@
+		}
+	}
+}
+
+function reapplySettings(): void {
+	select
+		.all(timelineItemSelector)
+		.forEach(element => processTimelineItem(element));
+}
+
+function restoreSettings(): void {
+	(select('#' + hideUnresolvedSelectorId) as HTMLInputElement)!.checked = CurrentSettings.HideUnresolved;
+	(select('#' + hideResolvedSelectorId) as HTMLInputElement)!.checked = CurrentSettings.HideResolved;
+	(select('#' + hideCommitsSelectorId) as HTMLInputElement)!.checked = CurrentSettings.HideCommits;
+	(select('#' + hideNormalCommentsSelectorId) as HTMLInputElement)!.checked = CurrentSettings.hideNormalComment;
+	(select('#' + hideOthersSelectorId) as HTMLInputElement)!.checked = CurrentSettings.HideOthers;
+	(select('#' + autoLoadHiddenSelectorId) as HTMLInputElement)!.checked = CurrentSettings.AutoLoadHidden;
+}
+
+function createItem(form: JSX.Element, id: string, title: string, summary: string, isSelected: boolean, hasTopBorder: boolean): void {
+	const element = (
+		<label className={'d-block p-3 ' + (hasTopBorder ? 'border-top' : '')}>
+			<div className="form-checkbox my-0">
+				<input id={id} type="checkbox" name="id" value="unsubscribe" checked={isSelected}/> {title}
+				<p className="note">
+					{summary}
+				</p>
+			</div>
+		</label>
+	);
+
+	form.append(element);
+}
+
+async function addTimelineItemsFilter(): Promise<void> {
+	const notifications = await elementReady(notiticationsSelector);
+	if (!notifications) {
+		return;
+	}
+
+	// Copy existing element and adapt its content
+	const timelineFilter = notifications.cloneNode(true);
+	timelineFilter.id = 'timeline-filters';
+
+	select('form.thread-subscribe-form', timelineFilter)!.remove();
+	const summary = select('summary', timelineFilter)!;
+	summary.setAttribute('aria-label', 'Customize timeline filters');
+	select('div.text-bold', summary)!.textContent = 'Filters';
+
+	createDetailsDialog(timelineFilter);
+	notifications.after(timelineFilter);
+
+	regenerateFilterSummary();
+}
+
+function createDetailsDialog(timelineFilter: Element): void {
+	const detailsDialog = select('details-dialog', timelineFilter)!;
+
+	detailsDialog.setAttribute('src', '');
+
+	detailsDialog.setAttribute('aria-label', 'Timeline filter settings');
+	select('div.Box-header h3', detailsDialog)!.textContent = 'Timeline filter settings';
+
+	// Close button should restore previous settings.
+	select('div.Box-header button', detailsDialog)!.addEventListener('click', restoreSettings);
+
+	const form = <div/>;
+
+	createItem(form, hideResolvedSelectorId, 'Hide resolved comments', '', CurrentSettings.HideResolved, false);
+	createItem(form, hideCommitsSelectorId, 'Hide commits', '', CurrentSettings.HideCommits, true);
+	createItem(form, hideUnresolvedSelectorId, 'Hide unresolved comments', '', CurrentSettings.HideUnresolved, true);
+	createItem(form, hideNormalCommentsSelectorId, 'Hide normal comments', 'Hides comments that does not contain unresolved/resolved state.', CurrentSettings.hideNormalComment, true);
```

Comment:
I think we can consider all comments as "unresolved" until they're hidden or resolved.

- Resolved/hidden comments
- Unresolved comments
- Commits (only on PRs)
- Meta events (instead of "other")
I think that by default all of these should be shown because hiding user content might cause confusion

Reply:
In my opinion it's very useful to have this special distinction for 3rd type of comments. It's very helpful in PRs with a lot of comments where you only want to see what unresolved problems you have left.

Look at PR that is a test url in this PR. Trying to find unresolved comments is pretty hard there, if there would be no option to just hide normal comments.

Changing name to meta events and not hiding anything by default is ok.

I will try to make those changes today/tomorrow.
Thanks for Review.

Reply:
Makes sense. Then maybe:
- Hidden comments (on Issues) | Resolved reviews (on PRs, even if it also targets hidden comments)
- Unresolved reviews (PR only)
- Regular comments
- Commits (PR only)
- Meta events

Reply:
Also perhaps commits should be rolled into the other events:

- Showing _only_ commits makes no sense, there's a dedicated view for that.
- Hiding commits but leaving other events also makes little sense IMHO

Reply:
Sorry, thinking out loud, but I had another idea 😅

What if was a single option (same display, except only one can be selected, like a `<select>`) among:

- [x] **Show all**
- [ ] Only show comments
  _Hides commits and events_
- [ ] Only show unresolved comments
  _Also hides resolved reviews and hidden comments_
- [ ] Only show unresolved reviews
  _Also hides regular comments (PR only)_

Reply:
Good ideas. I will implement them.

Reply:
I implemented the idea. The only thing left is hidden comments.